### DISCUSSION TITLE
local image starting with file:// won't work

### DISCUSSION THREAD

**Discussion by pierroo** [JimmyDaddy/react-native-image-marker#219] [Q&A] [answered]
**Describe the bug**
I recently upgraded RN to 0.73, so I had to upgrade this library to 1.2.6
But since then, markText with src image no longer works

I now get the error "please set image!"
when the image is indeed set.

**Devlopment environment(please complete the following information):**
 - react-native:  0.73.3
 - react-native-image-marker :  1.2.6

**Smartphone (please complete the following information):**
 - Device: emulator & device
 - OS: android (for now)

**Additional context**
Important to note: the image is coming from a viewshot capture, but still, that local file path used to work in previous versions.

EDIT: actually even with a local image using require('pathToImage') I still get the same error :/

**Comment by JimmyDaddy** [comment]
Hello @pierroo, can you provide me a sample code about how you use the library?

**Comment by pierroo** [comment]
Of course @JimmyDaddy , here it is:

```
viewShotRefs?.current
        .capture()
        .then(async uri => {
          console.log('view shot: ', uri);
          ImageMarker.markText({
            src: uri, // file://
            text: 'blablabla'
              }`,
            scale: 1,
            quality: 100,
            saveFormat: 'base64',
            position: 'bottomRight',
            color: '#ffffff',
            fontSize: 36,
            shadowStyle: {
              dx: 1,
              dy: 1,
              radius: 10,
              color: '#000000',
            },
          })
            .then(res => {
              goShare(res);
            })
            .catch(err => {
              console.log('error marking img: ', err);
            });
        })
        .catch(e => {
          console.log('could not capture full canvas: ', e);
        });
```

**Comment by JimmyDaddy** [comment, marked as answer]
@pierroo the API has been changed in v1.x.x, you should modify you code like below
```typescript
  viewShotRefs?.current
    .capture()
    .then(async (uri) => {
      console.log('view shot: ', uri);
      try {
        const markRes = await ImageMarker.markText({
          backgroundImage: {
            src: uri,
            scale: 1,
          },
          watermarkTexts: [
            {
              text: 'blablabla',
              position: {
                position: Position.bottomRight,
              },
              style: {
                color: '#ffffff',
                fontSize: 36,
                shadowStyle: {
                  dx: 1,
                  dy: 1,
                  radius: 10,
                  color: '#000000',
                },
              },
            },
          ],
          scale: 1,
          quality: 100,
          saveFormat: ImageFormat.base64,
        });
        goShare(markRes);
      } catch (err) {
        console.log('error marking img: ', err);
      }
    })
    .catch((e) => {
      console.log('could not capture full canvas: ', e);
    });

```
[document here](https://github.com/JimmyDaddy/react-native-image-marker?tab=readme-ov-file#example-4)

**Reply by pierroo** [reply]
thank you so much, it does work much better like that :)

**Reply by sushen123** [reply]
i tried to put base64 string in backgroundImage src but it give this message {"message": "java.lang.NullPointerException"}

**Reply by xyzuan** [reply]
i am too, do you have solution?
`
(NOBRIDGE) LOG  file:///data/user/0/com.acn.tm_app/cache/ImagePicker/10cf1f12-fc87-42c9-8910-6fc194bf2e5d.jpeg
(NOBRIDGE) ERROR  [Error: Exception in HostFunction: java.lang.NullPointerException]
`
